src/parquet/arrow/reader.cc

```diff
@@ -1467,16 +1075,21 @@ Status StructImpl::NextBatch(int batch_size, std::shared_ptr<Array>* out) {
   for (auto& child : children_) {
     std::shared_ptr<Array> child_array;
 
-    RETURN_NOT_OK(child->NextBatch(batch_size, &child_array));
-
+    RETURN_NOT_OK(child->NextBatch(records_to_read, &child_array));
     children_arrays.push_back(child_array);
   }
 
   RETURN_NOT_OK(DefLevelsToNullArray(&null_bitmap, &null_count));
 
-  *out = std::make_shared<StructArray>(field()->type(), batch_size, children_arrays,
-                                       null_bitmap, null_count);
+  int64_t struct_length = children_arrays[0]->length();
+  for (size_t i = 1; i < children_arrays.size(); ++i) {
+    if (children_arrays[i]->length() != struct_length) {
+      return Status::Invalid("Struct children had different lengths");
```

Comment:
Should this be a `DCHECK`? Or is it possible for this to happen because of error on the user's part?

Reply:
This would occur if the Parquet file is malformed somehow. I'll leave a TODO to consider this